### DISCUSSION TITLE
How do I use generate_series() with timestamptz?

### DISCUSSION THREAD

**Discussion by derhuerst** [duckdb/duckdb#7029] [Q&A] [answered]
- DuckDB v0.7.1 `b00b93f0b1`
- macOS 12.6.4 on ARM64 (Apple Silicon M2)

I'm trying to use the `generate_series(timestamptz, timestamptz, interval)` function, as [documented in the *ICU Timestamp Table Functions* section](https://duckdb.org/docs/sql/functions/timestamptz#icu-timestamp-table-functions), but it doesn't seem to "find" it?

```sql
LOAD icu;
SELECT unnest(range('2023-04-11T12:30+02'::timestamptz, '2023-04-11T13:30+02'::timestamptz, '20 minutes'::interval)) as t;
-- Error: Binder Error: No function matches the given name and argument types 'range(TIMESTAMP WITH TIME ZONE, TIMESTAMP WITH TIME ZONE, INTERVAL)'. You might need to add explicit type casts.
-- 	Candidate functions:
-- 	range(BIGINT) -> BIGINT[]
-- 	range(BIGINT, BIGINT) -> BIGINT[]
-- 	range(BIGINT, BIGINT, BIGINT) -> BIGINT[]
-- 	range(TIMESTAMP, TIMESTAMP, INTERVAL) -> TIMESTAMP[]
-- 
-- LINE 1: SELECT unnest(range('2023-04-11T12:30+02'::timestampt...
```

**Comment by Mytherin** [comment, marked as answer]
The `timestamptz` variant of the `range` function was added after the release of v0.7.1, the linked docs are for the latest development version. You need to use the pre-release to use that function (or wait for v0.8)
